### PR TITLE
add TestRunInValidCPUShares

### PR DESCRIPTION
**Pull request by sunyuan3**
Signed-off-by: Yuan Sun sunyuan3@huawei.com

### CODE REVIEW

File: integration-cli/docker_cli_run_unix_test.go [hunk 1]
```diff
@@ -374,3 +374,21 @@ func (s *DockerSuite) TestRunSwapLessThanMemoryLimit(c *check.C) {
 		c.Fatalf("Expected output to contain %q, not %q", out, expected)
 	}
 }
+
+func (s *DockerSuite) TestRunInValidCPUShares(c *check.C) {
```
Comment (aaronlehmann) [inline integration-cli/docker_cli_run_unix_test.go]: This should be called `TestRunInvalidCPUShares`.

**Comment by aaronlehmann** [comment]
The test looks good to me.

**Comment by cpuguy83** [comment]
I feel like there is already a test for this.

File: integration-cli/docker_cli_run_unix_test.go [hunk 2]
```diff
@@ -374,3 +374,21 @@ func (s *DockerSuite) TestRunSwapLessThanMemoryLimit(c *check.C) {
 		c.Fatalf("Expected output to contain %q, not %q", out, expected)
 	}
 }
+
+func (s *DockerSuite) TestRunInvalidCPUShares(c *check.C) {
+	testRequires(c, cpuShare)
+	out, _, err := dockerCmdWithError("run", "--cpu-shares", "1", "busybox", "echo", "test")
+	c.Assert(err, check.NotNil)
```
Comment (cpuguy83) [inline integration-cli/docker_cli_run_unix_test.go]: Can you add a `check.Commentf(out)` here? THis way if it fails we can see what happened on the failure.

File: integration-cli/docker_cli_run_unix_test.go [hunk 3]
```diff
@@ -374,3 +374,21 @@ func (s *DockerSuite) TestRunSwapLessThanMemoryLimit(c *check.C) {
...
+	expected := "The minimum allowed cpu-shares is 2"
+	c.Assert(out, checker.Contains, expected, check.Commentf("Expected output to contain %q, not %q", out, expected))
```
Comment (cpuguy83) [inline integration-cli/docker_cli_run_unix_test.go]: No need for the Commentf here, since that will be displayed in the output on failure anyway.

**Comment by cpuguy83** [comment]
But I couldn't find one, some nits

File: integration-cli/docker_cli_run_unix_test.go (hunk 2 above)
Comment (cpuguy83) [inline integration-cli/docker_cli_run_unix_test.go]: Same for the other tests

File: integration-cli/docker_cli_run_unix_test.go (hunk 3 above)
Comment (cpuguy83) [inline integration-cli/docker_cli_run_unix_test.go]: Same for the other tests.

**Comment by jfrazelle** [comment]
@sunyuan3 can you please update the pr per the feedback and i think we can move this forward, thanks :)

**Comment by sunyuan3** [comment]
Reopen it.

**Comment by sunyuan3** [comment]
Sorry for delay. The codes have been updated.

**Comment by cpuguy83** [comment]
LGTM

**Comment by vdemeester** [comment]
LGTM 🐶

**Pull request merged**